h/services/user_signup.py

```diff
@@ -95,11 +104,10 @@ def signup(self, require_activation=True, **kwargs):
                 # If the exception is not related to the email or username, re-raise it.
                 raise
 
-        # FIXME: this is horrible, but is needed until the
-        # notification/subscription system is made opt-out rather than opt-in
-        # (at least from the perspective of the database).
-        sub = Subscriptions(uri=user.userid, type="reply", active=True)
-        self.session.add(sub)
+        # Create default subscription entries for the user
+        self.subscription_service.set_subscriptions(
+            user_id=user.userid, subscriptions={}
+        )
```

Comment:
The `{}` here is a bit weird, because it's relying on side effects of the function to work.

We could do:

```python
self.subscription_service.set_subscriptions(
      user_id=user.userid, subscriptions={
            self.subscription_service.Types.REPLY: True
      }
  )
```

But if we add more we'll miss them so:

```python
self.subscription_service.set_subscriptions(
      user_id=user.userid, subscriptions={
            type_: True for type_ in self.subscription_service.Types.ALL_TYPES
      }
  )
```

Or we could have a `set_default_subscriptions` method?